core/services/pipeline/orm.go

```diff
@@ -127,6 +265,36 @@ func (o *orm) GetAllRuns() ([]Run, error) {
 	return runs, err
 }
 
+func (o *orm) GetUnfinishedRuns(now time.Time, fn func(run Run) error) error {
+	return sorm.Batch(func(offset, limit uint) (count uint, err error) {
+		var runs []Run
+
+		err = o.db.
+			Preload("PipelineSpec").
+			Preload("PipelineTaskRuns", func(db *gorm.DB) *gorm.DB {
+				return db.
+					Order("created_at ASC, id ASC")
+			}).
+			Where(`state = ? AND created_at < ?`, RunStatusRunning, now).
```

Comment:
I wonder if there should be an index on `state`?

Reply:
looks like we only use here, so seems unnecessary at the moment